timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java

```diff
@@ -280,15 +282,78 @@ public void putsCauseInvalidationsInSubsequentTransactions() {
     @Test
     public void leaderElectionFlushesCache() {
         putValue();
-        loadValue();
+        readValueAndAssertLoadedFromRemote();
 
         CLUSTER.failoverToNewLeader(Namespace.DEFAULT_NAMESPACE.getName());
         awaitTableWatched();
 
+        readValueAndAssertLoadedFromRemote();
+    }
+
+    @Test
+    public void leaderElectionDuringReadOnlyTransactionDoesNotCauseItToFail() {
+        putValue();
+        readValueAndAssertLoadedFromRemote();
+
         txnManager.runTaskThrowOnConflict(txn -> {
             assertThat(txn.get(TABLE_REF, ImmutableSet.of(CELL_1))).containsEntry(CELL_1, DATA_1);
+
+            // Calling these finalises the underlying cache, which normally means that no more operations should be
+            // carried out on the cache. However, since this cache should be cleared after the election, it is OK to
+            // do this earlier
+            assertHitValues(txn, ImmutableSet.of(TABLE_CELL_1));
+            assertLoadedValues(txn, ImmutableMap.of());
+
+            CLUSTER.failoverToNewLeader(Namespace.DEFAULT_NAMESPACE.getName());
```

Comment:
- Let's have this be a constant referenced here and by the creation path.
- I would strongly suggest using a random client as opposed to a "default" one. This Namespace here is a different concept, and also wouldn't play well with parallel tests doing the same thing.

Reply:
Makes sense